source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -360,8 +317,10 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
       ENVOY_STREAM_LOG(
           trace, "ext_authz filter set query parameter(s) on the request:", *decoder_callbacks_);
       for (const auto& [key, value] : response->query_parameters_to_set) {
-        ENVOY_STREAM_LOG(trace, "'{}={}'", *decoder_callbacks_, key, value);
-        modified_query_parameters->overwrite(key, value);
+        std::string encoded_key = Http::Utility::PercentEncoding::urlEncodeQueryParameter(key);
```

Comment:
Just curious why we are doing this. Same as below. places.

Reply:
Because query parameters get added to the path header, so if the query parameters are incorrectly encoded, the resulting path will also be incorrectly encoded.

Some kind of encoding / validation here is necessary, if for no other reason than adding CheckResponse to the ext_authz fuzzer input. The fuzzer will add invalid characters to the query parameters, which will then get added to the path, which will then cause an assert failure.

Reply:
Is there a bug in the original code which can justify the URL encoding change here?
If not, can we just do normal string checking here?